packages/core/src/node/uploadContribution.ts

```diff
@@ -0,0 +1,62 @@
+/*
+* Copyright (C) 2017 Ericsson and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+import { injectable, inject } from 'inversify';
+import { ILogger } from "../common";
+import { BackendApplicationContribution } from '../node';
+import * as express from 'express';
+import { Request } from 'express';
+import * as multer from 'multer';
+import * as fs from 'fs-extra';
+
+const storage = multer.diskStorage({
+    destination: function (req: Request, file, cb) {
+        let dest: string = req.body.dest;
+        if (dest === undefined) {
+            dest = "./uploads"; // Default value under: <current dir>/uploads
+        }
+        let finalDest = dest;
+        const fileURI = "file://";
+        if (dest.startsWith(fileURI)) {
+            finalDest = dest.slice(fileURI.length, dest.length);
+        }
+
+        // Create the folder if not existant
+        if (!fs.pathExistsSync(finalDest)) {
+            fs.mkdirpSync(finalDest);
+        }
+        cb(newFunction_1(), finalDest);
+    },
+    filename: function (req, file, cb) {
+        // Rename the file to its orinal name
+        cb(newFunction(), file.originalname);
+    }
+});
+const upload = multer({ storage: storage });
+
+@injectable()
+export class UploadContribution implements BackendApplicationContribution {
+    constructor(
+        @inject(ILogger) protected readonly logger: ILogger) {
+    }
+
+    configure(app: express.Application) {
+        this.logger.warn("Start configure the upload folder");
+
+        app.put('/upload', upload.array("file"),
+            function (req, res, next) {
+                res.send("Success\n");
+            });
+    }
+
+}
+function newFunction_1(): Error | null {
+    return null;
+}
+
+function newFunction(): Error | null {
```

Comment:
leftovers?

Reply:
Removed, use the callback with "null", no error handling